middlewares/refreshToken.ts

```diff
@@ -0,0 +1,16 @@
+import { NextFunction, Request, Response } from 'express';
+import config from '../utils/config';
+import { createAndSetJwtCookie } from '../utils/cookie';
+
+const refreshToken = (req: Request, res: Response, next: NextFunction): void => {
+  if (req.user) {
+    const now = (new Date()).getTime() / 1000;
+    if (req.user.exp - now > Number(config.refreshDurationHours) * 3600) {
+      createAndSetJwtCookie(res, req.user.psychologist, req.user.xsrfToken);
```

Comment:
No need to regenerate xsrfToken?

Reply:
i don't think so, the xsrfToken assure you that the person you give the cookie is the same as the one asking for the info. Here it just transfers it

Reply:
Yeah you're right, and `refreshToken` middleware is executed after `xsrfProtection` so it will always be a correct xsrf token at this point.